### DISCUSSION TITLE
Plugin not outputting text on load

### DISCUSSION THREAD

**Discussion by Arzte** [z-shell/zi#185] [Q&A] [answered]
Hiya! I'm having an issue getting this OMZ plugin https://github.com/oldratlee/hacker-quotes working in my migration from Zinit to ZI. It's dead simple and just outputs random text from a list before unloading, however, I've been unable to get it working for me. I'm pretty certain the issue is on my end, and It's not a bug, hence not opening an issue.

I've been using this to load it, and zi doesn't report any issues, but nothing pops up when it loads. (I have it at the bottom of my ``.zshrc``)
```bash
zi light oldratlee/hacker-quotes
```

**Comment by ss-o** [comment, marked as answer]
Hey :wave: 

I was unable to reproduce your issue:

```zsh
Downloading: oldratlee/hacker-quotes…
Cloning into '/home/hpuser/.zi/plugins/oldratlee---hacker-quotes'...
⠸ ░▒▓████████ OBJ: 100, PACK: 64/64, COMPR: 100%, REC: 100%, RES: 100%  
Compiling: hacker-quotes.plugin.zsh… ✔
❯ zsh -li
we set XMODIFIERS GTK_IM_MODULE QT_IM_MODULE in profile.d
It is not a language's weakness but its strengths that control the gradient of its change: Alas, a language never escapes its embryonic sac.
 - Alan J. Perlis
```

However, after checking the https://github.com/oldratlee/hacker-quotes it looks like the quote will be printed if Zsh interactive and login is true (https://github.com/oldratlee/hacker-quotes/blob/51acd6dc8b5e500f36eba1f0f5b89a79a27286d2/hacker-quotes.plugin.zsh#L4) and checks for `ZSH_HACKER_QUOTES_ENABLE_WHEN_INTERACTIVE`.

So basically if the plugin is installed with:

```zsh
zi light oldratlee/hacker-quotes
```

it should print the quote when running `zsh -il` and it should print output whenever `zsh` is run with:

```zsh
# I am guessing here.
zi ice atinit'ZSH_HACKER_QUOTES_ENABLE_WHEN_INTERACTIVE=true'
zi light oldratlee/hacker-quotes
```

```zsh
# Output when running: zsh
$ zsh
In man-machine symbiosis, it is man who must adjust: The machines can't.
 - Alan J. Perlis
```

Summarising the above if you are not getting the quote the problem should be in the basic config of your Zsh or environment setup in general. :octocat: 

Hope it helps :upside_down_face:

**Reply by Arzte** [reply]
Hmm, not sure why it needs the variable, outside of updates to my terminal changing how the default shell is launched, but it seems to have fixed it for me.

Thanks for the help!!

**Reply by ss-o** [reply]
No worries :+1:  

In a [Wiki](https://z.digitalclouds.dev/) you can find multiple methods to manage plugins, it's quite complex but very flexible and takes some time to get familiar with.

> Note: Soon I should be able to add additional info to the wiki, based on common issues/mistakes, from the public `.zshrc` of the users. Work in progress, but not far from being introduced.
